open-api/rest-catalog-open-api.yaml

```diff
@@ -2068,6 +2165,145 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        A flexible JSON object that contains information provided by the server,
+        to be utilized by clients for distributed planning. However it is recommended that servers
+        provide a plan-task-id, as well as a location to a manifest.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - data-file
+      properties:
+        data-file:
+          $ref: '#/components/schemas/ContentFile'
+        partition:
+          type: object
+          additionalProperties:
+            type: string
```

Comment:
This is not just a string map I think, it needs proper parsing

Reply:
At least it should not be 

```
          additionalProperties:
            type: string
```

And ideally we should match the behavior of the parser

Reply:
will rebase on drew pr

Reply:
Let's keep threads open until they are resolved.